src/v/cluster/controller.cc

```diff
@@ -548,23 +575,46 @@ ss::future<> controller::create_cluster() {
  * to it, and before we have started communicating with peers.
  */
 ss::future<> controller::cluster_creation_hook(cluster_discovery& discovery) {
+    const bool controller_log_exists = _raft0->last_visible_index()
+                                         > model::offset{}
+                                       || _raft0->config().brokers().size() > 1;
+
+    if (controller_log_exists && !_storage.local().get_cluster_uuid()) {
```

Comment:
I think we should move this into the `if (!is_cluster_founder)` condition instead.

Reply:
A cluster founder will never make it into this branch anyway because of `controller_log_exists`. However I agree that code paths will be clearer for a reader then.

Reply:
Yeah good point. Maybe we don't even need this check anymore.

Reply:
This code has been refactored significantly